Source/monster.cpp

```diff
@@ -2257,298 +2257,198 @@ void __fastcall M_StartFadeout(int i, int md, BOOL backwards)
 
 void __fastcall M_StartHeal(int i)
 {
-	int v1;            // edi
-	int v2;            // esi
-	CMonster *v3;      // eax
-	unsigned char *v4; // ecx
-	int v5;            // eax
-
-	v1 = i;
 	if ((DWORD)i >= MAXMONSTERS)
 		TermMsg("M_StartHeal: Invalid monster %d", i);
-	v2 = v1;
-	if (monster[v1].MType == NULL)
-		TermMsg("M_StartHeal: Monster %d \"%s\" MType NULL", v1, monster[v2].mName);
-	v3 = monster[v2].MType;
-	v4 = v3->Anims[MA_SPECIAL].Data[monster[v2]._mdir];
-	monster[v2]._mAnimData = v4;
-	v5 = v3->Anims[MA_SPECIAL].Frames;
-	monster[v2]._mFlags |= 2u;
-	monster[v2]._mAnimFrame = v5;
-	monster[v2]._mmode = MM_HEAL;
-	monster[v2]._mVar1 = monster[v2]._mmaxhp / (16 * (random(97, 5) + 4));
+	if (monster[i].MType == NULL)
+		TermMsg("M_StartHeal: Monster %d \"%s\" MType NULL", i, monster[i].mName);
+
+	MonsterStruct *Monst = &monster[i];
+	Monst->_mAnimData = Monst->MType->Anims[MA_SPECIAL].Data[Monst->_mdir];
+	Monst->_mAnimFrame = Monst->MType->Anims[MA_SPECIAL].Frames;
+	Monst->_mFlags |= 2;

```

Comment:
Do we have a define for mFlags?

Reply:
Not yet unfortunately, it's still not clear to me what it is being used for.